### PR TITLE
Add new example notebook for active learning

### PR DESCRIPTION
**Pull request by joelnkn**
Adding a notebook for active learning following the checklist in https://github.com/chemprop/chemprop/pull/772 (and https://github.com/chemprop/chemprop/issues/556).

### CODE REVIEW

**Comment by kevingreenman** [comment]
@joelnkn thanks for adding this! It looks like a great start to me at first glance. A couple notes:

- You mentioned on Slack a concern about the error metrics getting worse with more active learning iterations. I agree that this would normally be concerning, but I think in this case it might be caused by the very small size of the dataset; with only 100 total points in the best case, the active learning progression is very noisy, and the model is apparently getting lucky with the training points it chooses in early iterations of the model that lead to better metrics. We could consider using a larger dataset for this notebook, but we'll have to think about what makes sense because we don't want to unnecessarily bloat the github repo with large files.
- If we find a good dataset to use where the metrics decrease with more active learning iterations, it would be nice to make a plot at the end of the notebook to visualize this.
- Right now I think random is a good choice for the priority function (aka acquisition function). Once we add the uncertainty functionality, I think people would also like to see an example of how using uncertainty-based sampling might improve results more efficiently than sampling randomly.

**Comment by kevingreenman** [comment]
One other note: I suggest changing your `batch_size` variable name to something like `al_batch_size` to avoid confusion with the `batch_size` that's a hyperparameter in training the model without active learning.

File: examples/active_learning.ipynb [hunk 1]
```diff
@@ -0,0 +1,808 @@
...
+   "source": [
+    "# Active Learning\n",
+    "Active learning is an iterative process where a model actively selects the most informative data points to be labeled by an oracle (e.g. a human expert), optimizing the model's performance with fewer labeled samples. Active learning can be implemented with Chemprop through Python as demonstrated by this notebook."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import Packages"
```
Comment (shihchengli) [inline examples/active_learning.ipynb:15]:
```suggestion
    "### Import packages"
```

File: examples/active_learning.ipynb [hunk 2]
```diff
@@ -0,0 +1,808 @@
...
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
```
Comment (shihchengli) [inline examples/active_learning.ipynb:785]:
```suggestion
```

**Review by shihchengli — COMMENTED** [review]
The active learning seems not be very helpful in this test case. It is possible due to the size of the dataset, the validation sets are changed overtime, and the model hyperparameters. Can you comment this in the last cell of the notebook?

File: examples/active_learning.ipynb [hunk 3]
```diff
@@ -0,0 +1,808 @@
...
+    "\n",
+    "    result = trainer.test(mpnn, test_loader)\n",
+    "    results.append((len(trainval_data), result[0][\"batch_averaged_test/mse\"]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally we can view the results. The model's performance will hopefully improve with each iteration of active learning. Though this notebook is just an example. We didn't train the model for many epochs, and we used a very small dataset, so we don't expect to see the model improve. "
```
Comment (KnathanM) [inline examples/active_learning.ipynb:729]: @shihchengli What do you think about this explanation?

Comment (shihchengli) [reply examples/active_learning.ipynb:729]: It looks good!

**Review by shihchengli — APPROVED** [review]

**Pull request merged**